### PR TITLE
call get order item

### PR DESCRIPTION
**Pull request by samMcCarthyBJSS**
Story [AB#8119](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/8119)
Task [AB#7855](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/7855)

### CODE REVIEW

File: app/pages/sections/catalogue-solutions/routes.js [hunk 1]
```diff
@@ -45,8 +46,12 @@ export const catalogueSolutionsRoutes = (authProvider, addContext, sessionManage
   router.use('/select', catalogueSolutionsSelectRoutes(authProvider, addContext, sessionManager));
 
   router.get('/:orderItemId', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
-    const { orderId } = req.params;
+    const { orderId, orderItemId } = req.params;
     const accessToken = extractAccessToken({ req, tokenType: 'access' });
+
+    if (orderItemId) {
```
Comment (alibombaye) [inline app/pages/sections/catalogue-solutions/routes.js:52]: we will always have an orderItemId. however we only want to call getOrderItem if the orderItemId is not `newsolution`

File: app/pages/sections/catalogue-solutions/routes.test.js [hunk 2]
```diff
@@ -275,6 +279,7 @@ describe('catalogue-solutions section routes', () => {
 
     it('should redirect to /organisation/some-order-id/catalogue-solutions/newsolution if a recipient is selected', async () => {
       orderItemController.getSolution = jest.fn().mockResolvedValue({});
+      orderItemController.getOrderItem = jest.fn().mockResolvedValue({});
```
Comment (alibombaye) [inline app/pages/sections/catalogue-solutions/routes.test.js:282]: shouldn't be calling this in it's newsolution

File: app/pages/sections/catalogue-solutions/routes.js [hunk 3]
```diff
@@ -45,8 +46,12 @@ export const catalogueSolutionsRoutes = (authProvider, addContext, sessionManage
   router.use('/select', catalogueSolutionsSelectRoutes(authProvider, addContext, sessionManager));
 
   router.get('/:orderItemId', authProvider.authorise({ claim: 'ordering' }), withCatch(logger, authProvider, async (req, res) => {
-    const { orderId } = req.params;
+    const { orderId, orderItemId } = req.params;
     const accessToken = extractAccessToken({ req, tokenType: 'access' });
+
+    if (orderItemId !== 'newsolution') {
+      await getOrderItem(orderId, orderItemId, accessToken);
```
Comment (alibombaye) [inline app/pages/sections/catalogue-solutions/routes.js:53]: don't you need the braces? around the params

File: app/pages/sections/catalogue-solutions/routes.test.js [hunk 4]
```diff
@@ -180,6 +180,23 @@ describe('catalogue-solutions section routes', () => {
           expect(res.text.includes('data-test-id="error-title"')).toBeFalsy();
         });
     });
+
+    it.only('should return the catalogue-solutions order item page if authorised and is not a new solution', () => {
```
Comment (alibombaye) [inline app/pages/sections/catalogue-solutions/routes.test.js:184]: .only

File: app/pages/sections/catalogue-solutions/routes.js (hunk 3 above)
Comment (samMcCarthyBJSS) [reply app/pages/sections/catalogue-solutions/routes.js:53]: hm oh yeah
although probs could remove the brackets off the method in the controller too as they're already defined and getting passed in

Comment (alibombaye) [reply app/pages/sections/catalogue-solutions/routes.js:53]: I'd keep the braces just to be consistent with the rest

File: app/pages/sections/catalogue-solutions/routes.test.js (hunk 4 above)
Comment (alibombaye) [reply app/pages/sections/catalogue-solutions/routes.test.js:184]: the .only still here

**Pull request merged**